include/openssl/evp.h

```diff
@@ -691,20 +691,23 @@ OPENSSL_EXPORT int EVP_PKEY_keygen_init(EVP_PKEY_CTX *ctx);
 // containing the result. It returns one on success or zero on error.
 OPENSSL_EXPORT int EVP_PKEY_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY **out_pkey);
 
-OPENSSL_EXPORT int EVP_PKEY_encapsulate_init(EVP_PKEY_CTX *ctx,
-                                             const OSSL_PARAM params[]);
+// EVP_PKEY_encapsulate is operation defined for a KEM (Key Encapsulation
+// Mechanism). The function generates a shared secret and a ciphertext which
+// encapsulates the shared secret using the public key given within |ctx|.
 OPENSSL_EXPORT int EVP_PKEY_encapsulate(EVP_PKEY_CTX *ctx,
-                                        unsigned char *wrapped_key,
-                                        size_t *wrapped_key_len,
-                                        unsigned char *gen_key,
-                                        size_t *gen_key_len);
-OPENSSL_EXPORT int EVP_PKEY_decapsulate_init(EVP_PKEY_CTX *ctx,
-                                             const OSSL_PARAM params[]);
+                                        uint8_t *ciphertext,
+                                        size_t *ciphertext_len,
+                                        uint8_t *shared_secret_key,
+                                        size_t *shared_secret_len);
+
+// EVP_PKEY_decapsulate is operation defined for a KEM (Key Encapsulation
+// Mechanism). The function decapsulates the shared secret from the given
+// ciphertext using the private key given within |ctx|.
 OPENSSL_EXPORT int EVP_PKEY_decapsulate(EVP_PKEY_CTX *ctx,
```

Comment:
Same as for `EVP_PKEY_encapsulate`. In addition, document what happens if `shared_secret = NULL`.